packages/dds/merge-tree/src/client.ts

```diff
@@ -106,6 +107,10 @@ export class Client {
         this._mergeTree = new MergeTree(options);
     }
 
+    public createEndOfTreeSegment(): ISegment {
```

Comment:
i don't think we should expose this. it has a lot of caveats, but i'll keep an open mind as i continue through the pr

Reply:
ok. makes sense we might need this on root update. i might prefer we expose the root node as a getter, rather than creating this specific thing.

Reply:
main other reasonable solution I came up with is to refactor merge-tree to always retain the same root node. i haven't checked how easy that is to do.

if we do end up going this route, we should mark this API @internal at the very least; there's no reason it should be called outside of fluid repo and api mostly exists right now to shirk some layering concerns. we might find a better way to do that instead.

Reply:
an idea for a minimal change in this regard. EndOfTreeSegment would then track the root pointer, which should always be up to date, and keeps everything encapsulated in the package

``` diff
+export interface IRootMergeBlock extends IMergeBlock{
+    rootPointer?: {root: IRootMergeBlock}
+}
 /**
  * @deprecated For internal use only. public export will be removed.
  * @internal
@@ -413,7 +417,7 @@ export class MergeTree {
     private static readonly initBlockUpdateActions: BlockUpdateActions;
     private static readonly theUnfinishedNode = <IMergeBlock>{ childCount: -1 };

-    root: IMergeBlock;
+    private readonly rootPointer: NonNullable<IRootMergeBlock["rootPointer"]>;
     private readonly blockUpdateActions: BlockUpdateActions = MergeTree.initBlockUpdateActions;
     public readonly collabWindow = new CollaborationWindow();

@@ -435,7 +439,15 @@ export class MergeTree {

     // TODO: make and use interface describing options
     public constructor(public options?: PropertySet) {
-        this.root = this.makeBlock(0);
+        this.rootPointer = {root: this.makeBlock(0)};
+    }
+
+    public get root(){
+        return this.rootPointer.root;
+    }
+    public set root(value){
+        this.rootPointer.root = value;
+        value.rootPointer = this.rootPointer;
     }

     private makeBlock(childCount: number) {
```

Reply:
i also like the idea of making the root node readonly. not sure how hard that would be